protocols/v2/noise-sv2/src/aed_cipher.rs

```diff
@@ -1,16 +1,60 @@
+//! # AEAD Cipher Trait
+//!
+//! Defines the [`AeadCipher`] trait which standardizes the interface for Authenticated Encryption
+//! with Associated Data (AEAD) ciphers used within the Noise protocol implementation. The trait is
+//! implemented for two common cipher, [`ChaCha20Poly1305`] and [`Aes256Gcm`], providing encryption
+//! and decryption functionality with authenticated data.
+//!
+//! ## Overview
+//!
+//! AEAD ciphers provide both confidentiality and integrity by encrypting data and creating an
+//! authentication tag in a single step. The integrity of additional associated data (AAD) can also
+//! be verified without including it in the encrypted output.
+//!
+//! ## Usage
+//!
+//! The [`AeadCipher`] trait is used by the [`crate::handshake::HandshakeOp`] trait to perform
+//! cryptographic operations during the Noise protocol handshake, ensuring secure communication
+//! between parties.
+
 use aes_gcm::Aes256Gcm;
 use chacha20poly1305::{aead::Buffer, AeadInPlace, ChaCha20Poly1305, ChaChaPoly1305, KeyInit};
 
+/// Defines the interface for AEAD ciphers.
+///
+/// The [`AeadCipher`] trait provides a standard interface for initializing AEAD ciphers, and for
+/// performing encryption and decryption operations with additional AAD. This trait is implemented
```

Comment:
```suggestion
/// performing encryption and decryption operations with additional Authenticated Associated Data (AAD). This trait is implemented
```

Reply:
That makes sense... People might get confused when they see this.